oauth2client/client.py

```diff
@@ -1843,6 +1843,22 @@ def __init__(self, client_id,
         }
         self.params.update(kwargs)
 
+        self._check_for_approval_prompt()
```

Comment:
Do not call instance methods from the constructor of the instance.

`_check_for_approval_prompt` _barely_ uses `self`. I think in this constructor you should just change
```
self.params = {
    'access_type': 'offline',
    'response_type': 'code',
}
self.params.update(kwargs)

self._check_for_approval_prompt()
```
into
```
self.params = _oauth2_web_server_flow_params(kwargs)
```
.